File: August 2026/2026-08-10/26-remove-duplicates-from-sorted-array.py
# 26-remove-duplicates-from-sorted-array.py;aarray;syntax-help

class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        mp = {}

        for num in nums:
            mp[num] = mp.get(num, 0) + 1

        k = 0

        for num in sorted(mp):
            nums[k] = num
            k += 1

        return k

File: August 2026/2026-08-03/26-remove-duplicates-from-sorted-array.cpp
// 26 leetcode remove-duplicates-from-sorted-array.cpp;array-map;self\

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        map<int, int> mp;
        int n = nums.size(), k = 0;
        for(int i = 0; i < n; i++) {
            mp[nums[i]]++;
        }

        for(auto &it : mp) {
            nums[k++] = it.first;
        }

        return k;
    }
};